lakefile.lean

```diff
@@ -62,11 +62,21 @@ package mathlib where
 ## Mathlib libraries
 -/
 
+target nolints_style (pkg) : System.FilePath :=
+  inputTextFile <| pkg.dir / "scripts" / "nolints-style.txt"
```

Comment:
In fact, you could probably remove this one, right? `nolints-style.txt` is only used for `lake exe lint-style`, which is run as a separate executable by CI. (This is not related to syntax linters and caching their results.)

Leaving this in for paranoia is fine with me --- how about adding a comment that this is not defence in depth, and currently not actually required?

Reply:
Yes, I can; I removed it and added a comment explaining a little more about why.

In principle someone could start calling the function that reads `nolints-style.txt` from another linter, then suddenly we would need this again.

Probably it would be best to refactor things such that the script reads the file, and passes the result into the library function, to make it clear that the library can't read the file at import-time; but that's out of scope for this PR.

Reply:
:+1: Thanks!
I'm adding the `lint-style` refactoring to my list of things; expect a PR within 1-2 weeks. (Currently, some more urgent things have to be finished first. But this one should be very quick, once I start it.)

Reply:
In fact, I had some time earlier - #19384 implements this.